pandas/core/generic.py

```diff
@@ -5795,6 +5796,12 @@ def astype(
                 for i in range(len(self.columns))
             ]
 
+        elif is_categorical_dtype(self):
```

Comment:
this probably belongs on `core.internals.blocks.Block.astype`; `is_categorical_dtype(self)` doesnt make sense for DataFrame (also, for perf, always better to pass a dtype object to is_foo_dtype rather than an arraylike object)

Reply:
Right. Done (& noted re: dtype checks)